core/src/main/java/io/github/tcdl/Consumer.java

```diff
@@ -63,6 +69,14 @@ public void subscribe(MessageHandler messageHandler) {
         this.messageHandler = messageHandler;
     }
 
+    public void addSubscriber(String correlationId, MessageHandler messageHandler) {
+        messageHandlersByCorrelationId.put(correlationId, messageHandler);
+    }
+
+    public void removeSubscriber(String correlationId) {
```

Comment:
how can we now know that last interested collector received message and we can stop consuming?